### DISCUSSION TITLE
How to read etl file and dump all the events in it.

### DISCUSSION THREAD

**Discussion by heestree** [n4r1b/ferrisetw#89] [Q&A] [answered]
Hi,

Thanks for your ferrisetw crate. I am new to Rust and trying to read a etl file and dump all the events using ferrisetw. Can any expert show sample code for this? I am not sure if this is the right place to ask. If there is any other place to ask about ferrisetw, please let me know also.

**Comment by mstange** [comment]
This is #7, and, as far as I know, not supported yet in this crate. For now you can copy https://github.com/jrmuizel/etw-profiler/tree/main/etw-reader into your repo and call `open_trace` to read the etl file, see the code under etw-gecko in the same repo for an example.
If this is useful to you, maybe you can write a PR for this repo to add support for etl file reading.

**Comment by daladim** [comment, marked as answer]
I'm working on it, it should be possible to add this to the next release of ferrisetw

**Reply by daladim** [reply]
Well, that's harder than I planned :-/

See #7 for more details

**Reply by daladim** [reply]
That's finally ready! See #92
